source/client/client_worker_impl.cc

```diff
@@ -10,9 +14,12 @@ ClientWorkerImpl::ClientWorkerImpl(Envoy::Api::Api& api, Envoy::ThreadLocal::Ins
                                    Envoy::Stats::Store& store, const int worker_number,
                                    const Envoy::MonotonicTime starting_time,
                                    bool prefetch_connections)
-    : WorkerImpl(api, tls, store), worker_number_(worker_number), starting_time_(starting_time),
-      benchmark_client_(benchmark_client_factory.create(api, *dispatcher_, store_, std::move(uri),
-                                                        cluster_manager)),
+    : WorkerImpl(api, tls, store),
+      scope_(store_.createScope(fmt::format("worker.{}.benchmark.", worker_number))),
```

Comment:
You are doing the aggregation by hand so I suppose it is what works for you.  If you want to support aggregation by an outside system which uses tags-value pairs then it would make more sense to define a stat tag and export then with that format.   It isn't clear that it makes sense unless the workers are somehow unique (i.e. you added additional tags for say http/1 vs http/2 and had heterogenous workers).  Then perhaps it would be interesting to do the slicing on worker # via tags.  I don't know all the use cases so I can't say what the right thing is.

Reply:
Well, we also export the per-worker counters, e.g in yaml, it looks like:
https://gist.github.com/oschaaf/15c830c4a870a3ff23cf4141116ad955
(please disregard that the per-worker counts don't make sense, that's what this PR addresses, I collected this output with the version build from master)

Reply:
Then it depends on whether you want to export to a tag-based system.  Google3 (internal production) uses a tag-based system.

Reply:
I guess my feeling is why not treat it as a separate element?  What is the downside?

Reply:
There's no downside, so if you lean towards that, lets do it like that. 
Also - if it would be helpful, I think it's also pretty doable to add an output formatter which would just output out a flattened key/value list of everything.
(See https://github.com/envoyproxy/nighthawk/blob/master/source/client/output_collector_impl.cc, this could be done in a follow up or we can create an issue and prioritize it later)

Reply:
OK, so we are going to go with the separate element for worker.
I like the idea of the output collector.  Let's create an issue.  The rest looks good to me.

Reply:
Done in https://github.com/envoyproxy/nighthawk/pull/116/commits/f5eaf6908ccc073c5b49593fd87312b258ccb427; also filed: https://github.com/envoyproxy/nighthawk/issues/136